src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,42 @@
+import React, { Dispatch, SetStateAction, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  onTodoSelect: Dispatch<SetStateAction<Todo | null>>;
+  selectedUserId: number;
+  selectedTodo: Todo | null;
+};
+
+const defaultUser: User = {
+  id: 0,
+  name: 'Default Username',
+  email: 'Default Email',
+  phone: 'Default Phonenumber',
+};
+
+export const TodoModal: React.FC<Props> = ({
+  onTodoSelect,
+  selectedUserId,
+  selectedTodo,
+}) => {
+  const [user, setUser] = useState<User>(defaultUser);
+
+  useEffect(() => {
+    getUser(selectedUserId)
+      .then(setUser)
+      .catch(() => alert('Error while trying to load user!'));
+  }, [selectedUserId]);
+
+  const isLoaded = selectedTodo && user !== defaultUser;
```

Comment:
You're generating a key here on render. This can lead to unexpected behavior in your app. Check out the link in the checklist to understand why this is an issue.